### PR TITLE
Forcing user to update app

### PR DESCRIPTION
**Pull request by olafch**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Screenshot(s)

Please attach some screenshot or a movie if provided changes affects UI.

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] Add correct label to your pull request
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/AppLoading.tsx [hunk 1]
```diff
@@ -25,6 +28,41 @@ export const AppLoading: FC<PropsWithChildren> = ({ children }) => {
 
   const isLoading = !isLoadingComplete || isSignedIn === null || !isLayoutReady
 
+  const currentVersion = Application.nativeApplicationVersion || 'unknown'
+  const os = Platform.OS
+
+  const { mutate: checkForUpdate } = useSystemControllerCheckForAppUpdate()
+
```
Comment (Sworzen1) [inline src/components/AppLoading.tsx:35]: It can be custom hook, which check the update is necessary.

File: src/components/AppLoading.tsx [hunk 2]
```diff
@@ -25,6 +28,41 @@ export const AppLoading: FC<PropsWithChildren> = ({ children }) => {
 
   const isLoading = !isLoadingComplete || isSignedIn === null || !isLayoutReady
 
+  const currentVersion = Application.nativeApplicationVersion || 'unknown'
+  const os = Platform.OS
```
Comment (Sworzen1) [inline src/components/AppLoading.tsx:32]: destructuring, up to you

```suggestion
  const { OS } = Platform
```

File: src/components/AppLoading.tsx [hunk 3]
```diff
@@ -25,6 +28,41 @@ export const AppLoading: FC<PropsWithChildren> = ({ children }) => {
...
+            currentVersion,
+            os,
+          },
+        },
+        {
+          onSuccess: (data) => {
+            if (data.updateRequired) {
+              const storeUrl =
+                os === 'ios'
+                  ? `https://apps.apple.com/app/id${data.appId}`
```
Comment (Sworzen1) [inline src/components/AppLoading.tsx:50]: What do you think about move this and below url to consts ?

File: src/components/AppLoading.tsx [hunk 4]
```diff
@@ -25,6 +28,41 @@ export const AppLoading: FC<PropsWithChildren> = ({ children }) => {
...
+    if (os === 'ios' || os === 'android') {
+      checkForUpdate(
+        {
+          data: {
+            currentVersion,
+            os,
+          },
+        },
+        {
+          onSuccess: (data) => {
```
Comment (Sworzen1) [inline src/components/AppLoading.tsx:46]: destructuring

```suggestion
          onSuccess: ({ updateRequired }) => {
```

File: src/components/AppLoading.tsx [hunk 5]
```diff
@@ -25,6 +28,41 @@ export const AppLoading: FC<PropsWithChildren> = ({ children }) => {
 
   const isLoading = !isLoadingComplete || isSignedIn === null || !isLayoutReady
 
+  const currentVersion = Application.nativeApplicationVersion || 'unknown'
+  const os = Platform.OS
+
+  const { mutate: checkForUpdate } = useSystemControllerCheckForAppUpdate()
```
Comment (Sworzen1) [inline src/components/AppLoading.tsx:34]: Add isLoading from mutation to isLoading on 29 line

File: src/components/AppLoading.tsx (hunk 2 above)
Comment (olafch) [reply src/components/AppLoading.tsx:32]: I've tried but the swagger's schema includes os so while destructuring, is not working properly 

Captura de pantalla 20 05 2024 a 08 49 24 a  m

File: src/components/AppLoading.tsx (hunk 5 above)
Comment (olafch) [reply src/components/AppLoading.tsx:34]: Gonna add this, thank you 🤓

File: src/components/AppLoading.tsx (hunk 1 above)
Comment (olafch) [reply src/components/AppLoading.tsx:35]: Totally agree

File: src/components/AppLoading.tsx (hunk 3 above)
Comment (olafch) [reply src/components/AppLoading.tsx:50]: That's nice, thanks for the feedback, really appreciate it

File: src/components/AppLoading.tsx (hunk 2 above)
Comment (Sworzen1) [reply src/components/AppLoading.tsx]: In this case in data at the end point, can be: 
{
currentVersion,
os:OS
},
as I said, up to you

File: src/hooks/useCheckForAppUpdate.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import { useSystemControllerCheckForAppUpdate } from '@baca/api/query/system/system'
+import { useTranslation } from '@baca/hooks'
+import * as Application from 'expo-application'
+import { useEffect, useState } from 'react'
+import { Platform, Alert, Linking } from 'react-native'
+
+export const useCheckForAppUpdate = () => {
+  const { t } = useTranslation()
```
Comment (Sworzen1) [inline src/hooks/useCheckForAppUpdate.ts:8]: below line 7 stay only: code form line 8 and 13, rest of them: lines 9, 10 move outside the hook - above line 7,
lines 16, 17 move to const folder. Dont use setIsLoading(false), on every state of promise, read about onSettled.

Comment (olafch) [reply src/hooks/useCheckForAppUpdate.ts:12]: Updated, I'm really thankful for all these suggestions 😊

File: scripts/cli/constants.ts [hunk 7]
```diff
@@ -1,5 +1,7 @@
 export const CLI_ACTIONS = ['generate', 'g', 'bootstrap', 'b']
 
+export const APP_STORE_URL = `https://apps.apple.com/app/id`
+export const PLAY_STORE_URL = `https://play.google.com/store/apps/details?id=`
```
[the remark this answers is not in the document]
Comment (MateuszRostkowski) [reply scripts/cli/constants.ts:4]: orl `links.ts`

[the remark this answers is not in the document]
Comment (olafch) [reply scripts/cli/constants.ts]: Thank you Mateusz, I've just added this review 🤓

**Pull request merged**